include/zephyr/drivers/rtc.h

```diff
@@ -0,0 +1,502 @@
+ * alarm pending status will be cleared automatically. The alarm will
+ * remain enabled until manually disabled using
+ * \ref rtc_alarm_set_time().
+ *
+ * To disable the alarm callback for an alarm, the \p callback and
+ * \p user_data parameters must be set to NULL. When the alarm
+ * callback for an alarm is disabled, the alarm triggered event will
+ * set the alarm status to "pending". To check if the alarm status is
+ * "pending", use \ref rtc_alarm_is_pending().
+ *
+ * @param dev Device instance
+ * @param id Id of the alarm for which the callback shall be set
+ * @param callback Callback called when alarm occurs
+ * @param user_data Optional user data passed to callback
+ *
+ * @return 0 if successful
+ * @return -EINVAL if id is out of range
+ * @return -ENOTSUP if API is not supported by hardware
+ * @return -errno code if failure
+ */
+__syscall int rtc_alarm_set_callback(const struct device *dev, uint16_t id,
+				     rtc_alarm_callback callback, void *user_data);
+
+static inline int z_impl_rtc_alarm_set_callback(const struct device *dev, uint16_t id,
+						rtc_alarm_callback callback, void *user_data)
+{
+	const struct rtc_driver_api *api = (const struct rtc_driver_api *)dev->api;
+
+	return api->alarm_set_callback(dev, id, callback, user_data);
+}
+
+#endif /* CONFIG_RTC_ALARM */
+/**
+ * @}
+ */
+
+/**
+ * @brief RTC Interface Update
+ * @ingroup rtc_interface
+ * @{
+ */
+#if defined(CONFIG_RTC_UPDATE) || defined(__DOXYGEN__)
+
+/**
+ * @brief API for setting update callback.
+ *
+ * @details Setting the update callback will enable the update
+ * callback. The update callback will be invoked every time the
+ * RTC clock is updated by 1 second. It can be used to
+ * synchronize the RTC clock with other clock sources.
+ *
+ * To disable the update callback for the RTC clock, the
+ * \p callback and \p user_data parameters must be set to NULL.
+ *
+ * @param dev Device instance
+ * @param callback Callback called when update occurs
+ * @param user_data Optional user data passed to callback
+ *
+ * @return 0 if successful
+ * @return -ENOTSUP if API is not supported by hardware
+ * @return -errno code if failure
+ */
+__syscall int rtc_update_set_callback(const struct device *dev, rtc_update_callback callback,
+				      void *user_data);
+
+static inline int z_impl_rtc_update_set_callback(const struct device *dev,
+						 rtc_update_callback callback, void *user_data)
+{
+	const struct rtc_driver_api *api = (const struct rtc_driver_api *)dev->api;
+
+	return api->update_set_callback(dev, callback, user_data);
+}
+
+#endif /* CONFIG_RTC_UPDATE */
+/**
+ * @}
+ */
+
+/**
+ * @brief RTC Interface Calibration
+ * @ingroup rtc_interface
+ * @{
+ */
+#if defined(CONFIG_RTC_CALIBRATION) || defined(__DOXYGEN__)
+
+/**
+ * @brief API for setting RTC calibration.
+ *
+ * @details Calibration is applied to the RTC clock input
+ *
+ * @param dev Device instance
+ * @param calibration Calibration to set in parts per billion
```

Comment:
What are the semantics of this value? Does a positive value cause the clock to speed up or slow down? Should be clarified in the doc comment.

Reply:
Positive makes it faster, negative makes it slower, I will update the docs